DECIMER/config.py

```diff
@@ -107,10 +106,13 @@ def remove_transparent(image_path: str):
     ___
     Output: PIL.Image
     """
-    if pathlib.Path(image_path).suffix == ".HEIC":
-        png = HEIF_to_pillow(image_path)
-    else:
+    try:
         png = Image.open(image_path).convert("RGBA")
+    except Exception as e:
+        if type(e).__name__ == "UnidentifiedImageError":
+            png = HEIF_to_pillow(image_path)
+        else:
+            print("Error occured check image!")
```

Comment:
I think it would be a more comprehensive error if the error message was actually printed here. Knowing if there is a problem due to a FileNotFoundError something else could be useful for users.

Reply:
It does print the whole error with traceback:
Example:
```
Error occured check image!
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
/tmp/ipykernel_1404179/1665754835.py in <module>
----> 1 Predictor_exported.predict_SMILES(img_path)

/mnt/raid_drive/DECIMER_V1_Revision/Nature_methods/ChEMBL_31/New_model_decoder/Predictor_exported.py in predict_SMILES(image_path)
     78     try:
     79 
---> 80         chemical_structure = config.decode_image(image_path)
     81         predicted_tokens = DECIMER_V2(chemical_structure)
     82         predicted_SMILES = utils.decoder(detokenize_output(predicted_tokens))

/mnt/raid_drive/DECIMER_V1_Revision/Nature_methods/ChEMBL_31/New_model_decoder/config_old.py in decode_image(image_path)
    216         Processed image
    217     """
--> 218     img = remove_transparent(image_path)
    219     img = increase_contrast(img)
    220     img = get_bnw_image(img)

/mnt/raid_drive/DECIMER_V1_Revision/Nature_methods/ChEMBL_31/New_model_decoder/config_old.py in remove_transparent(image_path)
    135             print("Error occured check image!")
    136 
--> 137     background = Image.new("RGBA", png.size, (255, 255, 255))
    138 
    139     alpha_composite = Image.alpha_composite(background, png)

UnboundLocalError: local variable 'png' referenced before assignment

```

Reply:
This is not the original error! This error is a subsequent error. In the except block, the variable `png` is not assigned. No matter what the error is (for example a `FileNotFoundError`), the message will always be an `UnboundLocalError`!

Reply:
Now it should work